velox/type/TimestampConversion.h

```diff
@@ -44,6 +44,26 @@ constexpr const int32_t kMaxYear{292278994};
 constexpr const int32_t kYearInterval{400};
 constexpr const int32_t kDaysPerYearInterval{146097};
 
+// Enum to dictate parsing modes for date strings.
+//
+// kStrict: For date string conversion, align with DuckDB's implementation.
```

Comment:
Curious, why do we need parsing mode aligned with DuckDB? Do you happen to know where this is used?

Reply:
Good question. Looks like it's only used by the original date cast function that I replaced here in from_iso8601_date - `fromDateString()`. And this cast function is only used by the Date class' utility function `DateType::toDays()`.

As for the actual parsing, the only difference between `kStrict` and `kStandard` just seems that the former allows trailing whitespace at the end of the date string. Will need to test some more to see if this split is really necessary or if these modes can be consolidated. Maybe better to open that as a separate investigation, seems to be moving further away from the scope of this PR?